components/channel_info_rhs/menu.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {Constants} from 'utils/constants';
+
+const Icon = styled.div`
+    color: rgba(63, 67, 80, 0.56);
+`;
+
+interface MenuItemProps {
+    className?: string;
+
+    icon: JSX.Element;
+    text: string;
+    onClick: () => void;
```

Comment:
Nit: Consider making `onClick` an optional component and omit passing the noop function down below.

Reply:
the onClick will be mandatory, i just use the noop while I develop to avoid having my linter screaming at me because my tags are missing the onClck option haha!